### ISSUE TITLE
Improvement: Performance

### ISSUE THREAD

**Issue by cehartun** [hochschule-darmstadt/MetaAutoML#538]
**Reason (Why?)**
The current architecture gets slow with increasing amount of trainings

**Solution (What?)**

- Analyze and monitor the performance of the web application
- Find bottlenecks
- adjust MongoDB-Queries and architecture for better performance

**Out of scope**
Anything that is out of scope and shall not be implemented

**Relation to other Issues**
Provide additional context (link relevant issues)

**Acceptance criteria**
Describe acceptance criteria for testing (Don't repeat reason and solution again)

**Comment by simon-lw** [comment]
In the frontend the homepage seems to load the relevant information sequentially from the controller instead of in parallel.
The function to get recent trainings calls the function to get recent datasets which in turn calls the function to get home overview information. Due to this, the site will only finish loading after the function to get home overview information returns.
